resources/bundles/org.eclipse.core.filesystem/src/org/eclipse/core/internal/filesystem/zip/ZipFileStore.java

```diff
@@ -0,0 +1,559 @@
+		Path toDelete = null;
+		try (FileSystem zipFs = openZipFileSystem()) {
+			toDelete = zipFs.getPath(path.toString());
+			if (Files.exists(toDelete)) {
+				deleteRecursive(toDelete);
+			}
+		} catch (IOException | URISyntaxException e) {
+			throw new CoreException(new Status(IStatus.ERROR, "org.eclipse.core.filesystem.zip", "Error deleting file from zip: " + toDelete, e)); //$NON-NLS-1$ //$NON-NLS-2$
+		}
+	}
+
+	private void deleteRecursive(Path pathToDelete) throws IOException {
+		if (Files.isDirectory(pathToDelete)) {
+			// Use try-with-resources to close the directory stream automatically
+			try (Stream<Path> entries = Files.walk(pathToDelete)) {
+				// We need to sort it in reverse order so directories come after their contents
+				List<Path> sortedPaths = entries.sorted(Comparator.reverseOrder()).collect(Collectors.toList());
+				for (Path entry : sortedPaths) {
+					Files.delete(entry);
+				}
+			}
+		} else {
+			Files.delete(pathToDelete);
+		}
+	}
+
+	@Override
+	public IFileInfo fetchInfo(int options, IProgressMonitor monitor) throws CoreException {
+		try (FileSystem zipFs = openZipFileSystem()) {
+			Path zipEntryPath = zipFs.getPath(path.toString());
+			if (Files.exists(zipEntryPath)) {
+				BasicFileAttributes attrs = Files.readAttributes(zipEntryPath, BasicFileAttributes.class);
+				return convertToIFileInfo(zipEntryPath, attrs);
+			}
+		} catch (IOException | URISyntaxException e) {
+			throw new CoreException(new Status(IStatus.ERROR, getPluginId(), "Error accessing ZIP file", e)); //$NON-NLS-1$
+		}
+
+		// Correctly set up FileInfo before returning
+		FileInfo notFoundInfo = new FileInfo(path.lastSegment());
+		notFoundInfo.setExists(false);
+		return notFoundInfo;
+	}
+
+	/**
+	 * Finds the zip entry with the given name in this zip file. Returns the
+	 * entry and leaves the input stream open positioned at the beginning of the
+	 * bytes of that entry. Returns null if the entry could not be found.
+	 */
+	private ZipEntry findEntry(String name, ZipInputStream in) throws IOException {
+		ZipEntry current;
+		while ((current = in.getNextEntry()) != null) {
+			if (current.getName().equals(name)) {
+				return current;
+			}
+		}
+		return null;
+	}
+
+	@Override
+	public IFileStore getChild(String name) {
+		return new ZipFileStore(rootStore, path.append(name));
+	}
+
+	@Override
+	public String getName() {
+		String name = path.lastSegment();
+		return name == null ? "" : name; //$NON-NLS-1$
+	}
+
+	@Override
+	public IFileStore getParent() {
+		if (path.segmentCount() > 0) {
+			return new ZipFileStore(rootStore, path.removeLastSegments(1));
+		}
+		// the root entry has no parent
+		return null;
+	}
+
+	private String getPluginId() {
+		return FrameworkUtil.getBundle(this.getClass()).getSymbolicName();
+	}
+
+	public IPath getPath() {
+		return path;
+	}
+
+	private boolean isNested() {
+		return this.rootStore instanceof ZipFileStore;
```

Comment:
```suggestion
		return rootStore instanceof ZipFileStore;
```

Reply:
Note that unless there is a project rule emitting some warning here, it's all fine to keep `this.`.